File: Matlab/+ACME/Color/color_gradient.m
function [C] = color_gradient(CData,CRes,varargin)
% COLOR_GRADIENT  Creates a color gradients from the given color vector.
%   C = COLOR_GRADIENT(CData,CRes) returns a CResx3 color vector derived
%   from CData
CParam = linspace(0,1,row(CData))';
CRes   = linspace(0,1,CRes)';
C      = interp1(CParam,CData,CRes,varargin{:});
end

File: color/color_gradient.py
from ..utility.linspace import *
from .fetch_texture     import *


def color_gradient(cdata, cres=64):
    """
    Returns a gradient consisting of cres colors, from the input color data

    Parameters
    ----------
    cdata : Tensor
        a (N,3,) color tensor
    cres : int (optional)
        the number of output colors

    Returns
    -------
    Tensor
        a (cres,3,) color tensor
    """

    return fetch_texture1D(cdata, linspace(0, 1, cres))
